Eastern-Research-Group/mywaterway
# Feature/813 improve code coverage

## Related Issues:
* [HMW-813](https://jira.epa.gov/browse/HMW-813)

## Main Changes:
* Added more tests and removed unnecessary code in an effort to get code coverage above 80%.
* Made the `npm run coverage` command run both client and server tests.
* Finished switching over to the ArcGIS Online version of the attains GIS layers.
  * Had to update a couple of tests.
  * Had to refactor the logic for loading attains GIS data on the community page, to fix an issue with saving to AGO. 

## Steps To Test:
Running code coverage now takes an 1 hour and 15 minutes run. I'm ok with you just taking my word on this, but all tests passed. You can see the coverage in the HMW teams channel.

Below are the steps required to run the new tests for saving to ArcGIS Online.
1. Navigate to http://localhost:3000/community/dc/overview
2. Open the dev tools
3. Open the Add & Save Data widget
4. Provide a name and click "Save to ArcGIS Online"
5. Login to AGO
6. In the dev tools, go to the Application tab => Session Storage => localhost
7. Copy the contents of `esriJSAPIOAuth`
8. Create a `cypress.env.json` file under the main `app` folder of the repo
9. Paste the contents of `esriJSAPIOAuth` into the `cypress.env.json` file
10. Run code coverage or the `add-data-widget.cy.ts` tests
11. Don't forget to delete the `CYPRESS-TEST-HMW` items from your AGO account
     * Note: There is a test to help remind us to do this.
